src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -163,6 +164,11 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
 
             var settingsEditorFactory = this.ComponentModel.GetService<SettingsEditorFactory>();
             RegisterEditorFactory(settingsEditorFactory);
+
+            // Misc workspace has to be up and running by the time our package is usable so that it can track running
+            // doc events and appropriately map files to/from it and other relevant workspaces (like the
+            // metadata-as-source workspace).
+            await this.ComponentModel.GetService<MiscellaneousFilesWorkspace>().InitializeAsync(this, cancellationToken).ConfigureAwait(false);
```

Comment:
Did you want to do a await TaskScheduler.Default so the MEF creation is off the UI thread?